File: src/lib/relationships.ts
// Shared relationship type list (uzbek labels) — used in both bot keyboards and admin UI.
export const RELATIONSHIP_OPTIONS = [
  { value: "self", label: "Men (o'zim)" },
  { value: "father", label: "Otam" },
  { value: "mother", label: "Onam" },
  { value: "son", label: "O'g'lim" },
  { value: "daughter", label: "Qizim" },
  { value: "brother", label: "Akam/Ukam" },
  { value: "sister", label: "Opam/Singlim" },
  { value: "husband", label: "Erim" },
  { value: "wife", label: "Xotinim" },
  { value: "uncle_paternal", label: "Amakim" },
  { value: "uncle_maternal", label: "Tog'am" },
  { value: "aunt_paternal", label: "Ammam" },
  { value: "aunt_maternal", label: "Xolam" },
  { value: "cousin_male", label: "Amakivachcham (erkak)" },
  { value: "cousin_female", label: "Amakivachcham (ayol)" },
  { value: "grandfather", label: "Bobom" },
  { value: "grandmother", label: "Buvim" },
  { value: "grandson", label: "Nevaram (o'g'il)" },
  { value: "granddaughter", label: "Nevaram (qiz)" },
  { value: "father_in_law", label: "Qaynotam" },
  { value: "mother_in_law", label: "Qaynonam" },
  { value: "son_in_law", label: "Kuyovim" },
  { value: "daughter_in_law", label: "Kelinim" },
  { value: "brother_in_law", label: "Qaynaka/Qaynuka" },
  { value: "sister_in_law", label: "Qaynsingil" },
  { value: "nephew", label: "Jiyanim (o'g'il)" },
  { value: "niece", label: "Jiyanim (qiz)" },
  { value: "other", label: "Boshqa qarindosh" },
] as const;

export type RelationshipValue = typeof RELATIONSHIP_OPTIONS[number]["value"];

export function relationshipLabel(value: string | null | undefined): string {
  if (!value) return "—";
  return RELATIONSHIP_OPTIONS.find(r => r.value === value)?.label ?? value;
}

File: backend/src/MeningOilam.Application/Kinship/Relationships.cs
namespace MeningOilam.Application.Kinship;

/// <summary>Uzbek labels for relationship types (snake_case keys match the DB enum labels).</summary>
public static class Relationships
{
    public static readonly IReadOnlyList<(string Value, string Label)> Options = new (string, string)[]
    {
        ("self", "Men (o'zim)"), ("father", "Otam"), ("mother", "Onam"), ("son", "O'g'lim"),
        ("daughter", "Qizim"), ("brother", "Akam/Ukam"), ("sister", "Opam/Singlim"),
        ("husband", "Erim"), ("wife", "Xotinim"), ("uncle_paternal", "Amakim"),
        ("uncle_maternal", "Tog'am"), ("aunt_paternal", "Ammam"), ("aunt_maternal", "Xolam"),
        ("cousin_male", "Amakivachcham (erkak)"), ("cousin_female", "Amakivachcham (ayol)"),
        ("grandfather", "Bobom"), ("grandmother", "Buvim"), ("grandson", "Nevaram (o'g'il)"),
        ("granddaughter", "Nevaram (qiz)"), ("father_in_law", "Qaynotam"), ("mother_in_law", "Qaynonam"),
        ("son_in_law", "Kuyovim"), ("daughter_in_law", "Kelinim"), ("brother_in_law", "Qaynaka/Qaynuka"),
        ("sister_in_law", "Qaynsingil"), ("nephew", "Jiyanim (o'g'il)"), ("niece", "Jiyanim (qiz)"),
        ("step_father", "O'gay otam"), ("step_mother", "O'gay onam"), ("step_son", "O'gay o'g'lim"),
        ("step_daughter", "O'gay qizim"), ("half_brother", "O'gay akam/ukam"), ("half_sister", "O'gay opam/singlim"),
        ("great_grandfather", "Buvamning otasi"), ("great_grandmother", "Buvamning onasi"),
        ("great_grandson", "Chevaram (o'g'il)"), ("great_granddaughter", "Chevaram (qiz)"),
        ("godfather", "Otaxon"), ("godmother", "Onaxon"), ("family_friend", "Oila do'sti"),
        ("other", "Boshqa qarindosh"),
    };

    private static readonly Dictionary<string, string> _labels = Options.ToDictionary(o => o.Value, o => o.Label);

    public static string Label(string? value) =>
        value is null ? "—" : _labels.GetValueOrDefault(value, value);
}
